nx/lib/nx/random.ex

```diff
@@ -80,22 +80,33 @@ defmodule Nx.Random do
         [0, 12]
       >
 
-  A single key effectively consists of 64 bits, so all possible values
-  of a 64-bit integer result in a different key. However, when passing
-  an integer literal, Nx implicitly assumes its type to be `:s32`,
-  which would result in an overflow for large integers. Therefore,
-  when dealing with large seeds, make sure to explicitly use a 64 bit
-  type:
-
-      iex> Nx.Random.key(Nx.u64(999999999999))
+      iex> Nx.Random.key(999999999999)
       #Nx.Tensor<
         u32[2]
         [232, 3567587327]
       >
   """
-  defn key(seed) do
-    seed = Nx.as_type(seed, :u64)
+  deftransform key(seed) do
+    seed =
+      case seed do
+        seed when is_integer(seed) ->
+          Nx.u64(seed)
 
+        %Nx.Tensor{} = seed when seed.type == {:u, 64} ->
+          seed
+
+        %Nx.Tensor{} = seed when seed.type == {:s, 64} ->
+          Nx.bitcast(seed, {:u, 64})
```

Comment:
Do we really need this clause? No-one should be passing s64 now.

I am sorry, but I don't understand why we don't keep the `seed = Nx.as_type(seed, :u64)` version.

Reply:
for me, i would check for integer and convert it, otherwise cast to u64 inside the `key_n` function:

Reply:
Or do we also want to do bitcasts for f64?

Reply:
We do logical and with another `:u64`, so that's good.

Reply:
I added the `s64` case because there are explicit tests for this, and it has the same entropy as u64. Not sure about floats.

Reply:
Oh I misread the question. I think it's fair to assume seed is always an integer.

No preference on `{:s, 64}`, if there are tests, then yeah, bitcast sounds good.